src/main/java/Duke.java

```
@@ -1,10 +1,141 @@
+
+        stage.setScene(scene);
+        stage.show();
+
+        stage.setTitle("Duke");
+        stage.setResizable(false);
+        stage.setMinHeight(600.0);
+        stage.setMinWidth(400.0);
+
+        mainLayout.setPrefSize(400.0, 600.0);
+
+        scrollPane.setPrefSize(385, 535);
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        scrollPane.setVbarPolicy(ScrollPane.ScrollBarPolicy.ALWAYS);
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
+        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+
+        userInput.setPrefWidth(325.0);
+
+        sendButton.setPrefWidth(55.0);
+
+        AnchorPane.setTopAnchor(scrollPane, 1.0);
+
+        AnchorPane.setBottomAnchor(sendButton, 1.0);
+        AnchorPane.setRightAnchor(sendButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput , 1.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+        dialogContainer.getChildren().addAll(
+                DialogPanel.getDukeDialog(new Label(ui.initMessage()), new ImageView(bot))
+        );
+
+        // 3. Handle user input.
+        sendButton.setOnMouseClicked((event) -> {
+            handleUserInput();
+        });
+        userInput.setOnAction((event) -> {
+            if (handleUserInput()) {
+                Platform.exit();
+            };
+        });
+    }
+
+    private boolean handleUserInput() {
+        String response = "";
+        boolean isExit = false;
+        Label userText = new Label(userInput.getText());
+            try {
+                if(userInput.getText().contains("bye")) {
+                    isExit = true;
+                }
+                response = ui.readUserInput(userInput.getText());
+            } catch (Exception e) {
+                response = e.getMessage();
+            }
```

Indentation of the `try` block, and try not to catch all exceptions (but only specific ones)